src/main/java/org/javarosa/core/model/condition/Triggerable.java

```diff
@@ -65,9 +65,10 @@
      */
     private TreeReference contextRef;  //generic ref used to turn triggers into absolute references
 
-    // TODO Study why we really need this property. Looking at mutators, it should always equal the contextRef.
     /**
-     * The first context provided to this triggerable before reducing to the common root.
+     * The first context provided to this triggerable before reducing to the common root (see Triggerable.intersectContextWith).
+     * This is just the first bind encountered with the expression represented by this Triggerable so it's not clear how
+     * it's useful (and bind order for the same form definition affects its value).
      */
     private TreeReference originalContextRef;
```

Comment:
I don't think I understand what context is for at all

Reply:
When you have relative expressions, you need to be able to convert them to concrete nodes. The evaluation context lets you do that. So if you have the path foo and you’re in /data/repeat[10], you’re actually referring to /data/repeat[10]/foo. 

The problem here is that the “template” concept is represented in such a way that contextualization can lead you to lose that you want to refer to the template.